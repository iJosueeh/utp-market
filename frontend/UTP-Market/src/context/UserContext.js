
import React, { createContext, useState, useEffect } from 'react';

const UserContext = createContext(null);

// Helper function to get user from localStorage
const getStoredUser = () => {
    try {
        return JSON.parse(localStorage.getItem('user'));
    } catch (error) {
        return null;
    }
};

export const UserProvider = ({ children }) => {
    const [user, setUser] = useState(getStoredUser);
    const [token, setToken] = useState(() => localStorage.getItem('token'));
    const [loading, setLoading] = useState(false); // No longer loading on initial mount

    const login = (userData, authToken) => {
        localStorage.setItem('token', authToken);
        localStorage.setItem('user', JSON.stringify(userData));
        setToken(authToken);
        setUser(userData);
    };

    const logout = () => {
        localStorage.removeItem('token');
        localStorage.removeItem('user');
        setToken(null);
        setUser(null);
    };

    return (
        <UserContext.Provider value={{ user, token, login, logout, loading }}>
            {children}
        </UserContext.Provider>
    );
};

export default UserContext;
