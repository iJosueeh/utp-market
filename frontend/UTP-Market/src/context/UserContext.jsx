import React, { createContext, useState, useCallback, useMemo } from 'react';

const UserContext = createContext(null);

const getStoredUser = () => {
    try {
        return JSON.parse(localStorage.getItem('user'));
    } catch {
        return null;
    }
};

export const UserProvider = ({ children }) => {
    const [user, setUser] = useState(getStoredUser);
    const [token, setToken] = useState(() => localStorage.getItem('token'));
    const [loading] = useState(false);

    const login = useCallback((userData, authToken) => {
        localStorage.setItem('token', authToken);
        localStorage.setItem('user', JSON.stringify(userData));
        setToken(authToken);
        setUser(userData);
    }, []);

    const logout = useCallback(() => {
        localStorage.removeItem('token');
        localStorage.removeItem('user');
        setToken(null);
        setUser(null);
    }, []);

    const value = useMemo(() => ({
        user,
        token,
        login,
        logout,
        loading
    }), [user, token, login, logout, loading]);

    return (
        <UserContext.Provider value={value}>
            {children}
        </UserContext.Provider>
    );
};

export default UserContext;
