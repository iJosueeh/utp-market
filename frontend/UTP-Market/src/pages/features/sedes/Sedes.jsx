import React from "react";
export default function Sedes() {
  const sedes = [
    {
      id: 1,
      nombre: "Sede Norte",
      direccion: "Panamericana Norte, Av. Alfredo Mendiola 6377, Los Olivos 15306",
      telefono: "970804148",
      mapa: "https://www.google.com/maps/place/Universidad+Tecnol√≥gica+del+Per√∫/@-11.9530276,-77.0723372,1056m/data=!3m1!1e3!4m6!3m5!1s0x9105d1d877f532d7:0x8db19fe8e1f40feb!8m2!3d-11.9528899!4d-77.070192!16s%2Fg%2F11bwqpvckg?entry=ttu&g_ep=EgoyMDI1MDkyNC4wIKXMDSoASAFQAw%3D%3D"
    },
    {
      id: 2,
      nombre: "Sede Lima Centro",
      direccion: "Avenida Petit Thouars 116, Lima 15046",
      telefono: "(01) 3159600",
      mapa: "www.google.com/maps/place/Torre+Tecnologica+UTP+-+Sede+Hern√°n+Velarde/@-12.065257,-77.037297,19z/data=!4m5!3m4!1s0x9105c8ea518cc623:0xfdac3d89dfbfb056!8m2!3d-12.0653906!4d-77.0367775?shorturl=1&shorturl=1"
    },
    {
      id: 3,
      nombre: "Sede Sur",
      direccion: "Ctra. Panamericana Sur km 16, Villa EL Salvador 15842",
      telefono: "(01) 3159600",
      mapa: "https://www.google.com/maps/place/Universidad+Tecnol√≥gica+del+Per√∫+UTP/@-12.1936042,-76.9718558,528m/data=!3m1!1e3!4m6!3m5!1s0x9105b9a26836ac01:0x4fd233c0d43ca479!8m2!3d-12.1940306!4d-76.9714698!16s%2Fg%2F11hbqbtdnp?entry=ttu&g_ep=EgoyMDI1MDkyNC4wIKXMDSoASAFQAw%3D%3D"
    }
  ];

  return (
    <div className="container my-5">
      <h2 className="mb-4 text-center" style={{ color: "#B50D30" }}>
        üìç Nuestras Sedes
      </h2>

      <div className="row g-4">
        {sedes.map((sede) => (
          <div key={sede.id} className="col-12 col-md-6">
            <div className="card shadow-sm h-100">
              <div className="card-body">
                <h5 className="card-title" style={{ color: "#222222" }}>
                  {sede.nombre}
                </h5>
                <p className="card-text mb-1">
                  <strong>Direcci√≥n:</strong> {sede.direccion}
                </p>
                <p className="card-text mb-3">
                  <strong>Tel√©fono:</strong> {sede.telefono}
                </p>
                <div className="ratio ratio-16x9">
                  <iframe
                    src={sede.mapa}
                    style={{ border: 0 }}
                    allowFullScreen
                    loading="lazy"
                    title={sede.nombre}
                  ></iframe>
                </div>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}
