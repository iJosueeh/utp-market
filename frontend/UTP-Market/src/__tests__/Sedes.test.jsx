import React from 'react';
import { render, screen, within } from '@testing-library/react';
import { describe, test, expect, vi } from 'vitest';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from '../context/UserContext';
import Sedes from '../pages/features/sedes/Sedes';

// Mock a los componentes Navbar y Footer
vi.mock('../../../components/common/Navbar', () => ({
  default: () => <nav>Navbar</nav>
}));
vi.mock('../../../components/common/Footer', () => ({
  default: () => <footer>Footer</footer>
}));

const renderWithProviders = (ui) => {
  return render(
    <MemoryRouter>
      <UserProvider>
        {ui}
      </UserProvider>
    </MemoryRouter>
  );
};

describe('Sedes', () => {
  test('renders the main title', () => {
    renderWithProviders(<Sedes />);
    const titleElement = screen.getByText(/Nuestras Sedes/i);
    expect(titleElement).toBeInTheDocument();
  });

  test('renders all sede cards', () => {
    renderWithProviders(<Sedes />);
    const main = screen.getByRole('main');
    const sedeCards = within(main).getAllByRole('heading', { level: 5 });
    expect(sedeCards).toHaveLength(5);
  });

  test('renders sede names correctly', () => {
    renderWithProviders(<Sedes />);
    expect(screen.getByText('Sede Norte')).toBeInTheDocument();
    expect(screen.getByText('Sede Lima Centro')).toBeInTheDocument();
    expect(screen.getByText('Sede Lima Sur')).toBeInTheDocument();
    expect(screen.getByText('Sede San Juan de Lurigancho')).toBeInTheDocument();
    expect(screen.getByText('Sede Ate')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = renderWithProviders(<Sedes />);
    expect(asFragment()).toMatchSnapshot();
  });
});