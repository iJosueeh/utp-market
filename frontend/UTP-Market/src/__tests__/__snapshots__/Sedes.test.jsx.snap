// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Sedes > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex flex-column min-vh-100"
  >
    <nav
      class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top "
    >
      <div
        class="container-fluid px-0"
      >
        <div
          class="d-flex"
        >
          <a
            class="navbar-brand ms-5"
            href="/"
          >
            <img
              alt=""
              src="/src/assets/utpmarketlogo.png"
              style="height: 45px;"
            />
          </a>
        </div>
        <div
          class="order-lg-3"
        >
          <div
            class="d-flex align-items-center"
          >
            <div
              class="position-relative d-none d-lg-block"
            >
              <button
                class="btn btn-icon btn-dark rounded-circle bg-dark"
                type="button"
              >
                <i
                  class="bi bi-cart2 align-middle"
                />
              </button>
            </div>
            <span
              class="d-none d-lg-block"
            >
              <a
                class="btn btn-icon btn-dark rounded-circle bg-dark"
                href="#"
              >
                <i
                  class="bi bi-bell align-middle"
                />
              </a>
            </span>
            <div
              class="d-none d-lg-block"
            >
              <a
                class="btn btn-icon btn-dark rounded-circle bg-dark me-4 d-none d-lg-block"
                href="http://localhost:8080/login"
              >
                <i
                  class="bi bi-person align-middle"
                />
              </a>
            </div>
            <button
              aria-controls="navbar-default2"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler ms-2 me-3"
              data-bs-target="#navbar-default2"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
          </div>
        </div>
        <div
          class="collapse navbar-collapse justify-content-center"
          id="navbar-default2"
        >
          <ul
            class="navbar-nav mx-1 text-center"
          >
            <li
              class="nav-item mx-2"
            >
              <a
                class="nav-link text-white"
                data-discover="true"
                href="/main"
              >
                Inicio
              </a>
            </li>
            <li
              class="nav-item mx-2"
            >
              <a
                class="nav-link text-white"
                data-discover="true"
                href="/about-us"
              >
                Nosotros
              </a>
            </li>
            <li
              class="nav-item mx-2"
            >
              <a
                class="nav-link text-white"
                data-discover="true"
                href="/sedes"
              >
                Sedes
              </a>
            </li>
            <li
              class="nav-item mx-2"
            >
              <a
                class="nav-link text-white"
                data-discover="true"
                href="/help"
              >
                Ayuda
              </a>
            </li>
            <li
              class="nav-item mx-2"
            >
              <a
                aria-current="page"
                class="nav-link text-white "
                href="#"
              >
                Vender
              </a>
            </li>
            <li>
              <div
                class="dropdown d-none d-md-block"
              >
                <button
                  aria-expanded="false"
                  class="btn btn-light-primary text-white"
                  data-bs-toggle="dropdown"
                  id="dropdownMenuButton1"
                  type="button"
                >
                  Categoria
                  <i
                    class="bi bi-caret-down-fill ms-2 align-middle"
                  />
                </button>
                <ul
                  aria-labelledby="dropdownMenuButton1"
                  class="dropdown-menu rounded shadow-lg border-0 py-3"
                >
                  <li>
                    <a
                      class="dropdown-item px-4 py-2"
                      href="#"
                    >
                      Tutorías
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item px-4 py-2"
                      href="#"
                    >
                      Guías de estudio
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item px-4 py-2"
                      href="#"
                    >
                      Ventas de snack
                    </a>
                  </li>
                  <li>
                    <a
                      class="dropdown-item px-4 py-2"
                      href="#"
                    >
                      Materiales
                    </a>
                  </li>
                </ul>
              </div>
            </li>
            <li
              class="nav-item d-lg-none mt-3"
            >
              <div
                class="d-flex justify-content-center"
              >
                <a
                  class="btn btn-icon btn-dark rounded-circle bg-dark mx-2"
                  href="#"
                >
                  <i
                    class="bi bi-cart2 align-middle"
                  />
                </a>
                <a
                  class="btn btn-icon btn-dark rounded-circle bg-dark mx-2"
                  href="#"
                >
                  <i
                    class="bi bi-bell align-middle"
                  />
                </a>
                <a
                  class="btn btn-icon btn-dark rounded-circle bg-dark mx-2"
                  href="http://localhost:8080/login"
                >
                  <i
                    class="bi bi-person align-middle"
                  />
                </a>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <main
      class="container my-5 flex-grow-1 pt-5"
    >
      <h2
        class="mb-5 text-center fw-bold"
        style="color: rgb(181, 13, 48);"
      >
        Nuestras Sedes
      </h2>
      <div
        class="row g-4"
      >
        <div
          class="col-12 col-md-6 col-lg-4"
        >
          <div
            class="card shadow-sm h-100 border-0 rounded-3"
          >
            <div
              class="card-body text-center"
            >
              <h5
                class="card-title fw-bold mb-3"
                style="color: rgb(34, 34, 34);"
              >
                Sede Norte
              </h5>
              <p
                class="card-text mb-2"
              >
                <strong>
                  Dirección:
                </strong>
                 Panamericana Norte, Av. Alfredo Mendiola 6377, Los Olivos 15306
              </p>
              <p
                class="card-text mb-3"
              >
                <strong>
                  Teléfono:
                </strong>
                 970804148
              </p>
              <a
                href="https://www.google.com/maps/place/Panamericana+Norte+6377,+Los+Olivos+15306"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Imagen Sede Norte"
                  class="img-fluid rounded shadow-sm"
                  src="https://utp.edu.pe/sites/default/files/campus/campus-los-olivos-utp.webp"
                  style="max-height: 200px; object-fit: cover; width: 100%;"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-6 col-lg-4"
        >
          <div
            class="card shadow-sm h-100 border-0 rounded-3"
          >
            <div
              class="card-body text-center"
            >
              <h5
                class="card-title fw-bold mb-3"
                style="color: rgb(34, 34, 34);"
              >
                Sede Lima Centro
              </h5>
              <p
                class="card-text mb-2"
              >
                <strong>
                  Dirección:
                </strong>
                 Avenida Petit Thouars 116, Lima 15046
              </p>
              <p
                class="card-text mb-3"
              >
                <strong>
                  Teléfono:
                </strong>
                 (01) 3159600
              </p>
              <a
                href="https://www.google.com/maps/place/Avenida+Petit+Thouars+116,+Lima+15046"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Imagen Sede Lima Centro"
                  class="img-fluid rounded shadow-sm"
                  src="https://utp.edu.pe/sites/default/files/campus/utp-lima-centro.webp"
                  style="max-height: 200px; object-fit: cover; width: 100%;"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-6 col-lg-4"
        >
          <div
            class="card shadow-sm h-100 border-0 rounded-3"
          >
            <div
              class="card-body text-center"
            >
              <h5
                class="card-title fw-bold mb-3"
                style="color: rgb(34, 34, 34);"
              >
                Sede Lima Sur
              </h5>
              <p
                class="card-text mb-2"
              >
                <strong>
                  Dirección:
                </strong>
                 Ctra. Panamericana Sur km 16, Villa El Salvador 15842
              </p>
              <p
                class="card-text mb-3"
              >
                <strong>
                  Teléfono:
                </strong>
                 (01) 3159600
              </p>
              <a
                href="https://www.google.com/maps/place/Panamericana+Sur+Km+16,+Villa+El+Salvador+15842"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Imagen Sede Lima Sur"
                  class="img-fluid rounded shadow-sm"
                  src="https://utp.edu.pe/sites/default/files/campus/LimaSur-.webp"
                  style="max-height: 200px; object-fit: cover; width: 100%;"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-6 col-lg-4"
        >
          <div
            class="card shadow-sm h-100 border-0 rounded-3"
          >
            <div
              class="card-body text-center"
            >
              <h5
                class="card-title fw-bold mb-3"
                style="color: rgb(34, 34, 34);"
              >
                Sede San Juan de Lurigancho
              </h5>
              <p
                class="card-text mb-2"
              >
                <strong>
                  Dirección:
                </strong>
                 Av. Wiesse 571, San Juan de Lurigancho 15434
              </p>
              <p
                class="card-text mb-3"
              >
                <strong>
                  Teléfono:
                </strong>
                 (01) 3159600
              </p>
              <a
                href="https://maps.app.goo.gl/REhYfwjyYBLGhErc6"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Imagen Sede San Juan de Lurigancho"
                  class="img-fluid rounded shadow-sm"
                  src="https://utp.edu.pe/sites/default/files/campus/SJL-.webp"
                  style="max-height: 200px; object-fit: cover; width: 100%;"
                />
              </a>
            </div>
          </div>
        </div>
        <div
          class="col-12 col-md-6 col-lg-4"
        >
          <div
            class="card shadow-sm h-100 border-0 rounded-3"
          >
            <div
              class="card-body text-center"
            >
              <h5
                class="card-title fw-bold mb-3"
                style="color: rgb(34, 34, 34);"
              >
                Sede Ate
              </h5>
              <p
                class="card-text mb-2"
              >
                <strong>
                  Dirección:
                </strong>
                 Av. Carretera Central 123, Ate, Lima
              </p>
              <p
                class="card-text mb-3"
              >
                <strong>
                  Teléfono:
                </strong>
                 (01) 3159600
              </p>
              <a
                href="https://maps.app.goo.gl/m6p3aiWzAuyf8mLu7"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt="Imagen Sede Ate"
                  class="img-fluid rounded shadow-sm"
                  src="https://utp.edu.pe/sites/default/files/campus/utp-sede-ate.webp"
                  style="max-height: 200px; object-fit: cover; width: 100%;"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="py-lg-8 py-5 bg-dark text-white"
    >
      <div
        class="container "
      >
        <div
          class="row text-center align-items-center text-md-start"
        >
          <div
            class="col-12 col-md-4 mb-4 mb-md-0 "
          >
            <h5
              class="fw-bold mb-3"
            >
              Contactanos
            </h5>
            <div
              class="d-grid gap-2"
            >
              <span>
                <i
                  class="bi bi-whatsapp me-2"
                >
                   +51 922 873 004
                </i>
              </span>
              <span>
                <i
                  class="bi bi-telephone me-2"
                >
                   +51 922 873 004 
                </i>
              </span>
              <span>
                <i
                  class="bi bi-envelope-at me-2"
                >
                   UTP-Market@outlook.com
                </i>
              </span>
            </div>
          </div>
          <div
            class="col-12 col-md-4 mb-4 mb-md-0 text-center"
          >
            <img
              class="mb-3"
              src="/src/assets/utpmarketlogo.png"
              style="height: 45px;"
            />
          </div>
          <div
            class="col-12 col-md-4 text-center text-md-end"
          >
            <div
              class="my-6"
            >
              <a
                aria-label="YouTube"
                class="me-4"
                href="https://www.youtube.com/"
                rel="noopener noreferrer"
                target="_blank"
                title="YouTube"
              >
                <i
                  class="bi bi-youtube fs-4 text-white"
                />
              </a>
              <a
                aria-label="Facebook"
                class="me-4"
                href="https://www.facebook.com/"
                rel="noopener noreferrer"
                target="_blank"
                title="Facebook"
              >
                <i
                  class="bi bi-facebook fs-4 text-white"
                />
              </a>
              <a
                aria-label="Twitter"
                class="me-4"
                href="https://twitter.com/"
                rel="noopener noreferrer"
                target="_blank"
                title="Twitter"
              >
                <i
                  class="bi bi-twitter fs-4 text-white"
                />
              </a>
              <a
                aria-label="Instagram"
                class="me-4"
                href="https://www.instagram.com/"
                rel="noopener noreferrer"
                target="_blank"
                title="Instagram"
              >
                <i
                  class="bi bi-instagram fs-4 text-white"
                />
              </a>
              <a
                aria-label="LinkedIn"
                class=""
                href="https://www.linkedin.com/"
                rel="noopener noreferrer"
                target="_blank"
                title="LinkedIn"
              >
                <i
                  class="bi bi-linkedin fs-4 text-white"
                />
              </a>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-12 text-center mt-4"
            >
              <span>
                 © UTP MARKET  2025. Todos los derechos reservados
              </span>
            </div>
          </div>
        </div>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
